src/lib/autocomplete/autocomplete.spec.ts

```diff
@@ -1478,6 +1478,54 @@ describe('MatAutocomplete', () => {
           .toEqual(Math.floor(panelBottom), `Expected panel to stay aligned after filtering.`);
     }));
 
+    it('should fall back to above position when requested if options are added while ' +
+        'the panel is open', fakeAsync(() => {
+      let fixture = createComponent(AutocompleteWithOnPushDelay);
+      fixture.detectChanges();
+
+      let inputEl = fixture.debugElement.query(By.css('input')).nativeElement;
+      let inputReference = fixture.debugElement.query(By.css('.mat-form-field-flex')).nativeElement;
+
+      // Push the element down so it has a little bit of space, but not enough to render.
+      inputReference.style.bottom = '10px';
+      inputReference.style.position = 'fixed';
+
+      // Focus the input to load the deferred options.
+      dispatchFakeEvent(inputEl, 'focusin');
+      tick(1000);
+
+      fixture.detectChanges();
+      tick();
+
+      Promise.resolve().then(() => {
```

Comment:
Having promises in tests either leads to unreliable tests since the callback doesn't end up being executed, or flaky tests since some browsers end up throttling them while they're minimized. You can take a look at how we've handled similar cases in the other tests. My guess is that you need a `zone.simulateZoneExit()` call in there.

Reply:
@crisbeto Thanks, I actually based this test off the other one that uses `AutocompleteWithOnPushDelay`. That also uses a promise, but I'll try to avoid it in this spec.